plugins/modules/ndo_ipsla_monitoring_policy.py

```diff
@@ -0,0 +1,504 @@
+        mso.fail_json(msg="Invalid value provided for sla_frequency: {0}; it must be in the range 1 - 300".format(sla_frequency))
+
+    if detect_multiplier is not None and detect_multiplier not in range(1, 101):
+        mso.fail_json(msg="Invalid value provided for detect_multiplier: {0}; it must be in the range 1 - 100".format(detect_multiplier))
+
+    if request_data_size is not None and request_data_size not in range(0, 17514):
+        mso.fail_json(msg="Invalid value provided for request_data_size: {0}; it must be in the range 0 - 17513".format(request_data_size))
+
+    if type_of_service is not None and type_of_service not in range(0, 256):
+        mso.fail_json(msg="Invalid value provided for type_of_service: {0}; it must be in the range 0 - 255".format(type_of_service))
+
+    if operation_timeout is not None and sla_frequency is not None and operation_timeout > (sla_frequency * 1000):
+        mso.fail_json(
+            msg="Invalid value provided for operation_timeout: {0}; must be less than or equal to: {1}".format(operation_timeout, (sla_frequency * 1000))
+        )
+
+    if threshold is not None and operation_timeout is not None and threshold > operation_timeout:
+        mso.fail_json(
+            msg="Invalid value provided for threshold: {0}; it must be less than or equal to the operation_timeout: {1}".format(threshold, operation_timeout)
+        )
+
+    if ipv6_traffic_class is not None and ipv6_traffic_class not in range(0, 256):
+        mso.fail_json(msg="Invalid value provided for ipv6_traffic_class: {0}; it must be in the range 0 - 255".format(ipv6_traffic_class))
+
+    if sla_type == "http":
+        destination_port = 80
+
+    if sla_type != "http":
+        if http_version or http_uri:
+            mso.fail_json(msg="http_version and http_uri can only be used when sla_type is 'http'")
+
+    ops = []
+    match = None
+
+    mso_template = MSOTemplate(mso, "tenant", template)
+    mso_template.validate_template("tenantPolicy")
+    object_description = "IPSLA Monitoring Policy"
+
+    path = "/tenantPolicyTemplate/template/ipslaMonitoringPolicies"
+    existing_ipsla_policies = mso_template.template.get("tenantPolicyTemplate", {}).get("template", {}).get("ipslaMonitoringPolicies", [])
+    if ipsla_monitoring_policy or ipsla_monitoring_policy_uuid:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_ipsla_policies,
+            [KVPair("uuid", ipsla_monitoring_policy_uuid) if ipsla_monitoring_policy_uuid else KVPair("name", ipsla_monitoring_policy)],
+        )
+        if match:
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = existing_ipsla_policies
+
+    if state == "present":
+        if match:
+            if ipsla_monitoring_policy and match.details.get("name") != ipsla_monitoring_policy:
+                ops.append(
```

Comment:
@sajagana  wrote a function for this, asked him to do a separate PR for it so we can rebase the open PRs on it. Shoudl we do this?
https://github.com/CiscoDevNet/ansible-mso/pull/563/files#diff-f3e3720de5c754cfeb714aa1ed0f9d0a1a043f8c0005944d58565decefe1817bR31

Reply:
Does this help in any way?

Reply:
I think the function is helpful to reduce complexity and code duplication.
However, maybe we can start using once its merged in instead of having to rebase everything?

Reply:
Done!